pino.js

```diff
@@ -59,36 +63,32 @@ function pino (opts, stream) {
     iopts.level = 'silent'
   }
 
-  var logger = new Pino(iopts, istream)
-  if (iopts.cache) {
-    // setImmediate is causing a very weird crash:
-    //    Assertion failed: (cb_v->IsFunction()), function MakeCallback...
-    // but setTimeout isn't *shrug*
-    setTimeout(function () {
-      if (!tools.streamIsBlockable(istream)) {
-        logger.emit('error', new Error('stream must have a file descriptor in extreme mode'))
-      }
-    }, 100)
-
-    events.onExit(function (code, evt) {
-      var buf = iopts.cache.buf
-      if (buf) {
-        // We need to block the process exit long enough to flush the buffer
-        // to the destination stream. We do that by forcing a synchronous
-        // write directly to the stream's file descriptor.
-        var fd = (istream.fd) ? istream.fd : istream._handle.fd
-        fs.writeSync(fd, buf)
-      }
-      if (!process._events[evt] || process._events[evt].length < 2 || !process._events[evt].filter(function (f) {
-        return f + '' !== events.onExit.passCode + '' && f + '' !== events.onExit.insertCode + ''
```

Comment:
this logic should be moved to the new code, because this was preventing `process.exit` in case of an installed `'SIGHUP'` handler. We might also use https://nodejs.org/api/events.html#events_emitter_listenercount_eventname to count those, as we are support 4+ now.

@davidmarkclements what this big check matching strings was doing?